databricks_cli/configure/provider.py

```diff
@@ -49,7 +49,14 @@ def _create_section_if_absent(raw_config, profile):
 
 
 def _get_option_if_exists(raw_config, profile, option):
-    return raw_config.get(profile, option) if raw_config.has_option(profile, option) else None
+    if profile == DEFAULT_SECTION:
+        # We must handle the DEFAULT_SECTION differently since it is not in the _sections property
+        # of raw config.
+        return raw_config.get(profile, option) if raw_config.has_option(profile, option) else None
+    # Check if option is defined in the profile.
+    elif not option in raw_config._sections.get(profile, {}).keys():
```

Comment:
using `_sections` is a little hacky since it is not a public API. We should use `sections()` and `items()`:

```python
elif profile in raw_config.sections() and option in dict(raw_config.items(profile)):
  return raw_config.get(profile, options)
else
  return None
```

Reply:
This won't work unfortunately since `items` returns values from the default section. For example, in this conf
```
[DEFAULT]
host = https://logfood.cloud.databricks.com
token = dapiREDACTED

[apple]
host = https://logfood.cloud.databricks.com
username = a
password = a
```
`raw_config.items()` is 
`[('host', 'https://logfood.cloud.databricks.com'), ('token', 'dapi15c319c96bb6497f655723cbc6706e8b'), ('username', 'a'), ('password', 'a')]`.